### ISSUE TITLE
SSML api blocking

### ISSUE THREAD

**Issue by zhzLuke96** [lenML/Speech-AI-Forge#207]
有人反馈 SSML api 会卡住，webui合成不会，大概定位是 BatchSynth.start_generate_async 有问题
并且这个问题可能和性能提升需求有关 #192 

## 截图

![Image](https://github.com/user-attachments/assets/bd3bcf07-a97b-4e48-9709-cea906051371)

现象就是无报错 blocking，接口无返回hang住

**Comment by YanyanPop** [comment]
linux环境 ubuntu11.4
会创建speech.mp3但音频片段长度为0，无内容，想问下这种情况是不是也是卡住了？
$ python launch.py
fatal: No names found, cannot describe anything.
2025-03-11 09:07:48,735 - datasets - INFO - PyTorch version 2.5.1 available.
2025-03-11 09:07:49,521 - modules.core.models.tts.CosyVoiceModel - INFO - No CosyVoice model found
2025-03-11 09:07:49,802 - modules.devices.devices - INFO - Using half precision: torch.float16
2025-03-11 09:07:49,803 - modules.devices.devices - INFO - Using device: cuda
2025-03-11 09:07:50,184 - modules.api.Api - INFO - Registered API: GET /v1/ping
2025-03-11 09:07:50,185 - modules.api.Api - INFO - Registered API: GET /v1/versions
2025-03-11 09:07:50,185 - modules.api.Api - INFO - Registered API: GET /v1/audio_formats
2025-03-11 09:07:50,186 - modules.api.Api - INFO - Registered API: GET /v1/models/reload
2025-03-11 09:07:50,187 - modules.api.Api - INFO - Registered API: GET /v1/models/unload
2025-03-11 09:07:50,187 - modules.api.Api - INFO - Registered API: GET /v1/models/list
2025-03-11 09:07:50,188 - modules.api.Api - INFO - Registered API: GET /v1/styles/list
2025-03-11 09:07:50,190 - modules.api.Api - INFO - Registered API: GET /v1/speakers/list
2025-03-11 09:07:50,191 - modules.api.Api - INFO - Registered API: POST /v1/speakers/refresh
2025-03-11 09:07:50,192 - modules.api.Api - INFO - Registered API: POST /v1/speakers/update
2025-03-11 09:07:50,193 - modules.api.Api - INFO - Registered API: POST /v1/speaker/create
2025-03-11 09:07:50,194 - modules.api.Api - INFO - Registered API: POST /v1/speaker/update
2025-03-11 09:07:50,195 - modules.api.Api - INFO - Registered API: POST /v1/speaker/detail
2025-03-11 09:07:50,204 - modules.api.Api - INFO - Registered API: GET /v1/tts
2025-03-11 09:07:50,205 - modules.api.Api - INFO - Registered API: POST /v1/ssml
2025-03-11 09:07:50,207 - modules.api.Api - INFO - Registered API: POST /v1/text:synthesize
2025-03-11 09:07:50,207 - modules.api.Api - INFO - Registered API: POST /v1/speech:recognize
2025-03-11 09:07:50,208 - modules.api.Api - INFO - Registered API: POST /v1/speech:longrunningrecognize
2025-03-11 09:07:50,210 - modules.api.Api - INFO - Registered API: POST /v1/audio/speech
2025-03-11 09:07:50,215 - modules.api.Api - INFO - Registered API: POST /v1/audio/transcriptions
2025-03-11 09:07:50,216 - modules.api.Api - INFO - Registered API: POST /v1/prompt/refine
2025-03-11 09:07:50,217 - modules.api.Api - INFO - Registered API: POST /v1/text/normalize
2025-03-11 09:07:50,218 - modules.api.Api - INFO - Registered API: GET /v1/xtts_v2/speakers
2025-03-11 09:07:50,219 - modules.api.Api - INFO - Registered API: POST /v1/xtts_v2/tts_to_audio
2025-03-11 09:07:50,220 - modules.api.Api - INFO - Registered API: GET /v1/xtts_v2/tts_stream
2025-03-11 09:07:50,221 - modules.api.Api - INFO - Registered API: POST /v1/xtts_v2/set_tts_settings
2025-03-11 09:07:50,232 - modules.api.Api - INFO - Registered API: POST /v1/stt/transcribe
2025-03-11 09:07:50,241 - modules.api.Api - INFO - Registered API: POST /v1/stt/stream
2025-03-11 09:07:50,246 - modules.api.Api - INFO - Registered API: POST /v1/vc
2025-03-11 09:07:50,248 - modules.api.Api - INFO - Registered API: POST /v2/tts
2025-03-11 09:07:50,248 - modules.api.api_setup - INFO - allow CORS origin: *
INFO:     Started server process [525378]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:7870 (Press CTRL+C to quit)
2025-03-11 09:07:56,614 WETEXT INFO found existing fst: /home/gxy/anaconda3/envs/speech_ai_forge/lib/python3.10/site-packages/tn/en_tn_tagger.fst
2025-03-11 09:07:56,614 - wetext-en_normalizer - INFO - found existing fst: /home/gxy/anaconda3/envs/speech_ai_forge/lib/python3.10/site-packages/tn/en_tn_tagger.fst
2025-03-11 09:07:56,614 WETEXT INFO                     /home/gxy/anaconda3/envs/speech_ai_forge/lib/python3.10/site-packages/tn/en_tn_verbalizer.fst
2025-03-11 09:07:56,614 - wetext-en_normalizer - INFO -                     /home/gxy/anaconda3/envs/speech_ai_forge/lib/python3.10/site-packages/tn/en_tn_verbalizer.fst
2025-03-11 09:07:56,614 WETEXT INFO skip building fst for en_normalizer ...
2025-03-11 09:07:56,614 - wetext-en_normalizer - INFO - skip building fst for en_normalizer ...
2025-03-11 09:07:57,416 - modules.core.models.tts.ChatTTS.ChatTTS - INFO - Loading ChatTTS models
2025-03-11 09:07:57,437 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - try to load from local: ./models/ChatTTS
2025-03-11 09:07:57,437 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - checking assets...
2025-03-11 09:07:58,361 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - all assets are already latest.
2025-03-11 09:07:58,857 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - vocos loaded.
2025-03-11 09:07:59,056 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - dvae loaded.
2025-03-11 09:08:02,820 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - gpt loaded.
2025-03-11 09:08:03,077 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - decoder loaded.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
2025-03-11 09:08:03,091 - modules.repos_static.ChatTTS.ChatTTS.core - INFO - tokenizer loaded.
2025-03-11 09:08:03,113 - modules.core.models.tts.ChatTTS.ChatTTS - INFO - ChatTTS models loaded
code:  11%|█████████████████▌                                                                                                                                               | 223/2048(max) [00:03, 58.17it/s]
2025-03-11 09:08:08,941 - modules.core.handler.encoder.StreamEncoder - INFO - StreamEncoder header set, channels: 1, sample_width: 2, sample_rate: 24000
调用指令：
curl http://localhost:7870/v1/audio/speech   -H "Authorization: Bearer anything_your_wanna"   -H "Content-Type: application/json"   -d '{
    "model": "chattts",
    "input": "Today is a wonderful day to build something people love! [lbreak]",
    "voice": "female2",
    "style": "chat"
  }'   --output speech.mp3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  222k    0  222k  100   155   198k    138  0:00:01  0:00:01 --:--:--  198k

**Comment by zhzLuke96** [comment]
@YanyanPop 我测试了一下，感觉应该和这个issues无关，只是保存文件后缀名的问题 😂

是这样：
- 最新版本，默认的输出格式是最快的无需编码的 PCM 流，也就是 wav 格式
- 但是 api 的 markdown 文档还没改，默认不应该是 `speech.mp3` 应该是 `speech.wav` ，或者配置 `response_format` 参数

所以，你要测试的话可以这样:
```
curl http://localhost:7870/v1/audio/speech \
  -H "Authorization: Bearer anything_your_wanna" \
  -H "Content-Type: application/json" \
  -d '{
    "model": "chattts",
    "input": "Today is a wonderful day to build something people love! [lbreak]",
    "voice": "female2",
    "style": "chat",
    "response_format": "mp3"
  }' \
  --output speech.mp3
```
或者这样
```
curl http://localhost:7870/v1/audio/speech \
  -H "Authorization: Bearer anything_your_wanna" \
  -H "Content-Type: application/json" \
  -d '{
    "model": "chattts",
    "input": "Today is a wonderful day to build something people love! [lbreak]",
    "voice": "female2",
    "style": "chat"
  }' \
  --output speech.wav
```

稍等我会简单更新一下文档

**Comment by zhzLuke96** [comment]
这个 issues 基本已经在这个 [commit](https://github.com/lenML/Speech-AI-Forge/commit/6333e0989faa7289dcec79117d497b39e63fa5fc) 修复了
issues 我关闭了，如果有其他类似问题开新issues吧
